File: Whishpermate/WhisperMateShared/Models/DictionaryEntry.swift
import Foundation

public struct DictionaryEntry: Identifiable, Codable {
    public let id: UUID
    public var trigger: String
    public var replacement: String?
    public var isEnabled: Bool

    public init(id: UUID = UUID(), trigger: String, replacement: String? = nil, isEnabled: Bool = true) {
        self.id = id
        self.trigger = trigger
        self.replacement = replacement
        self.isEnabled = isEnabled
    }
}

File: AIDictationAndroid/app/src/main/java/com/whispermate/aidictation/domain/model/DictionaryEntry.kt
package com.whispermate.aidictation.domain.model

import com.squareup.moshi.JsonClass
import java.util.UUID

@JsonClass(generateAdapter = true)
data class DictionaryEntry(
    val id: String = UUID.randomUUID().toString(),
    val trigger: String,
    val replacement: String? = null,
    val isEnabled: Boolean = true
)
